src/main/kotlin/no/nav/syfo/infrastructure/database/VurderingRepository.kt

```diff
@@ -13,57 +15,49 @@ import java.util.UUID
 class VurderingRepository(private val database: DatabaseInterface) : IVurderingRepository {
     override fun createForhandsvarsel(
         pdf: ByteArray,
-        document: List<DocumentComponent>,
-        personident: PersonIdent,
-        veilederident: String,
-        begrunnelse: String,
+        vurdering: Vurdering,
     ) {
         database.connection.use { connection ->
-            val vurdering = connection.createVurdering(
-                personIdent = personident,
-                veileder = veilederident,
-                begrunnelse = begrunnelse,
-            )
-            val varsel = connection.createVarsel(
-                vurderingId = vurdering.id,
-                document = document,
+            val pVurdering = connection.createVurdering(vurdering)
+            val pVarsel = connection.createVarsel(
+                vurderingId = pVurdering.id,
+                varsel = vurdering.varsel
+                    ?: throw IllegalStateException("Vurdering should have Varsel when creating forhåndsvarsel"),
```

Comment:
Ikke veldig krise, men tror ikke jeg er kjempefan av å kaste exception inne i argumentene i et metodekall 🤔 Og er det riktig å opprette en vurdering i databasen av `Vurdering`en når den ikke inneholder alt den skal? Dette er jo tross alt `createForhandsvarsel`, da ville jeg tenkt at vurderinger som ikke er gyldige forhåndsvarsler må bruke en annen metode for å lagres i databasen.

Reply:
Ja, enig, tror jeg skal endre dette 👍🏼 https://github.com/navikt/isarbeidsuforhet/pull/20#discussion_r1502700058